php-checks/src/main/java/org/sonar/php/checks/UseOfUninitializedVariableCheck.java

```diff
@@ -264,4 +338,99 @@ private static Tree skipParentArrayAccess(Tree tree) {
     return child;
   }
 
+  private static boolean uninitializedVariableDeclaration(VariableIdentifierTree tree) {
+    return (tree.getParent().is(Kind.VARIABLE_DECLARATION) &&
+      ((VariableDeclarationTree) tree.getParent()).equalToken() == null);
+  }
+
+  private static class UninitializedUsageFindVisitor extends SummaryUpdateVisitor {
```

Comment:
It's weird to me to have `UninitializedUsageFindVisitor extends SummaryUpdateVisitor`.
I would expect `UninitializedUsageFindVisitor` to not modify its `blockSummary`.